extensions/ql-vscode/src/view/common/DataGrid.tsx

```diff
@@ -0,0 +1,79 @@
+import * as React from "react";
+import { styled } from "styled-components";
+
+/*
+ * A drop-in replacement for the VSCodeDataGrid family of components.
+ *
+ * The difference is that the `display: grid` styling is applied to `DataGrid`, whereas
+ * in the VS Code version that styling is applied to `VSCodeDataGridRow`. This gives
+ * column alignment across rows in situation with dynamic contents. It also allows
+ * for cells to span multiple rows and all the other features of data grids.
+ */
+
+const StyledDataGrid = styled.div<{ $gridTemplateColumns: string | number }>`
+  display: grid;
+  grid-template-columns: ${(props) => props.$gridTemplateColumns};
+  padding: calc((var(--design-unit) / 4) * 1px) 0;
+  box-sizing: border-box;
+  width: 100%;
+  background: transparent;
+`;
+
+interface DataGridProps {
+  gridTemplateColumns: string | number;
+  children: React.ReactNode;
+}
+
+export function DataGrid(props: DataGridProps) {
+  const { gridTemplateColumns, children } = props;
+
+  return (
+    <StyledDataGrid
+      aria-label="DataGrid"
+      $gridTemplateColumns={gridTemplateColumns}
+    >
+      {children}
+    </StyledDataGrid>
+  );
+}
+
+export const DataGridRow = styled.div<{ $focused?: boolean }>`
```

Comment:
Is there a particular reason for choosing to prefix the prop name with `$`? Could we simply use `focused` instead?

Reply:
I see now that these are transient props. Could we ensure that we don't need to use transient props outside the `DataGrid.tsx` file, for example by creating a wrapper component? I think it's fine to use these for props that can only be used from within the file.

Reply:
Probably either works, but making it a transient prop could a cleaner. It probably doesn't matter in this case though, so I'm happy with whatever is cleaner.

Sadly, I'm finding that if you omit a prop on a styled component or add extra unknown ones it doesn't cause a build/linting error like it does for regular components. Perhas it would be safer to leave out the `$` prefix where unneeded, so the prop is named how people expect.

For others reading this, see https://github.com/github/vscode-codeql/pull/2992 for more info.